packages/typescript/src/browser/typescript-callhierarchy-service.ts

```diff
@@ -32,8 +33,15 @@ export class TypeScriptCallHierarchyService extends AbstractDefaultCallHierarchy
      * are returned as a reference as well. As these are not calls they have to be filtered.
      * We also just want ot see the top-most caller symbol.
      */
-    async getEnclosingCallerSymbol(reference: Location, context: CallHierarchyContext): Promise<SymbolInformation | undefined> {
-        const symbols = (await context.getAllSymbols(reference.uri)).filter(s => this.isCallable(s));
+    async getEnclosingCallerSymbol(reference: Location, context: CallHierarchyContext): Promise<ExtendedDocumentSymbol | SymbolInformation | undefined> {
+        const allSymbols = await context.getAllSymbols(reference.uri);
+        if (allSymbols.length === 0) {
+            return undefined;
+        }
+        if (DocumentSymbol.is(allSymbols[0])) {
+            return this.getEnclosingRootSymbol(reference, context);
```

Comment:
I think the hierarchy of symbols let us find the enclosing one very safely, right?